### ISSUE TITLE
`HungarianMatcher` token cost matrix is transposed inconsistently

### ISSUE THREAD

**Issue by zoeadams68** [Yuliang-Liu/MonkeyOCRv2#67]
# `HungarianMatcher` token cost matrix is transposed inconsistently

**Severity:** Medium
**File:** `formula/cdm_local/modules/visual_matcher.py:92-128`

The `calculate_token_cost` method returns `token_cost.T`, producing a
matrix of shape `(num_gt, num_pred)`. However, the indexing convention
throughout the rest of the matcher uses `(gt, pred)` rows and columns:

```python
def calculate_token_cost(self, box_gt, box_pred):
    # ... builds pred_token_logits of shape (num_pred, num_classes)
    # ... builds gt_token_array of shape (num_gt,)
    token_cost = 1.0 - pred_token_logits[:, gt_token_array]  # (num_pred, num_gt)
    norm_token_cost = 1.0 - norm_pred_token_logits[:, norm_gt_token_array]
    token_cost[np.logical_and(token_cost==1, norm_token_cost==0)] = 0.005
    return token_cost.T  # now (num_gt, num_pred)
```

The position cost (`calculate_l1_cost`) returns `(num_gt, num_pred)`
naturally. But the `0.005` soft-match replacement is applied *before*
the transpose, meaning it modifies the `(num_pred, num_gt)` matrix.
After transpose, the positions of modified entries are correct, but
the semantic meaning of "pred matches gt with normalized token" is
applied to the wrong axis pair.

### Why it matters
When a predicted token matches a GT token only after normalization
(e.g. `\leq` vs `\le`), the cost is set to 0.005 on the `(pred, gt)`
axis. After transpose this becomes `(gt, pred)` which is correct for
the final cost matrix. However, the boolean condition
`token_cost==1 AND norm_token_cost==0` tests the *un-transposed*
matrix where rows are predictions and columns are GTs. This is
semantically correct but fragile -- any future refactoring that
changes the transpose order will silently break the matching.
